sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1605,7 +1605,7 @@ class AdaptiveQueryExecSuite
   }
 
   test("SPARK-33933: Materialize BroadcastQueryStage first in AQE") {
-    val testAppender = new LogAppender("aqe query stage materialization order test")
+    val testAppender = new LogAppender("aqe query stage materialization order test", 10000)
```

Comment:
cc Max @MaxGekk since this max is from Max 
(https://github.com/apache/spark/commit/88fc8dbc09c5d24ae89413ab1e1fbabdf1fd8028#diff-1c74a76903c7da8f8424992b46b2f99157609726bc580d60e2d0858ea11c2aecR197) :D

Reply:
Do you know what kind of debug logs we have, and why so much. Maybe reduce them?

Reply:
I try running this locally, the log size is 388.
```
loggerName.map(Logger.getLogger).getOrElse(Logger.getRootLogger)
```
The test is flaky since it is using `Logger.getRootLogger`.
Shall we increase the default limit of LogAppender instead? What was the reason for adding a limit in the LogAppender?

Reply:
> What was the reason for adding a limit in the LogAppender?

I added the limit to prevent OOM and memory pressure in tests. It was added when I troubleshooted random OOMs in tests.

Reply:
I guess many debug logs are introducing after https://github.com/apache/spark/pull/31269 ?

Reply:
That only added one debug log when materializing a stage, and a query won't have many stages.

Reply:
I mean other feature's debug logs can also be collected by this LogAppender